packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -920,10 +916,276 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
     }
   }
 
+  @protected
+  @override
+  RevealedOffset getOffsetToReveal(
+    RenderObject target,
+    double alignment, {
+    Rect? rect,
+    Axis? axis,
+  }) {
+    // We must know which axis we are revealing for, since Offset refers to only
+    // one of two scroll positions.
+    assert(axis != null);
+
+    late final double offset;
+    late final AxisDirection axisDirection;
+    (offset, axisDirection) = switch (axis!) {
+      Axis.vertical => (verticalOffset.pixels, verticalAxisDirection),
+      Axis.horizontal => (horizontalOffset.pixels, horizontalAxisDirection),
+    };
+
+    rect ??= target.paintBounds;
+    // `child` will be the last RenderObject before the viewport when walking
+    // up from `target`.
+    RenderObject child = target;
+    while (child.parent != this) {
+      child = child.parent!;
+    }
+
+    assert(child.parent == this);
+    final RenderBox box = child as RenderBox;
+    final Rect rectLocal = MatrixUtils.transformRect(target.getTransformTo(child), rect);
+
+    final double targetMainAxisExtent;
+    double leadingScrollOffset = offset;
+    // The scroll offset of `rect` within `child`.
+    switch (axisDirection) {
+      case AxisDirection.up:
+        leadingScrollOffset += child.size.height - rectLocal.bottom;
+        targetMainAxisExtent = rectLocal.height;
+      case AxisDirection.right:
+        leadingScrollOffset += rectLocal.left;
+        targetMainAxisExtent = rectLocal.width;
+      case AxisDirection.down:
+        leadingScrollOffset += rectLocal.top;
+        targetMainAxisExtent = rectLocal.height;
+      case AxisDirection.left:
+        leadingScrollOffset += child.size.width - rectLocal.right;
+        targetMainAxisExtent = rectLocal.width;
+    }
+
+    // The scroll offset in the viewport to `rect`.
+    final TwoDimensionalViewportParentData childParentData = parentDataOf(box);
+    leadingScrollOffset += switch (axisDirection) {
+      AxisDirection.down => childParentData.paintOffset!.dy,
+      AxisDirection.up => viewportDimension.height - childParentData.paintOffset!.dy - box.size.height,
+      AxisDirection.right => childParentData.paintOffset!.dx,
+      AxisDirection.left => viewportDimension.width - childParentData.paintOffset!.dx - box.size.width,
+    };
+
+    // This step assumes the viewport's layout is up-to-date, i.e., if
+    // the position is changed after the last performLayout, the new scroll
```

Comment:
Is this something we should fix later? or it is something that shouldn't happen. If it is the former, we should add a TODO

Reply:
I don't think we can, at least, it is the same in 1D, this is a copied comment. Maybe @LongCatIsLooong can speak more to why? I thought it was because we are about to performLayout anyways with a new offset to reveal, so if the scroll position change while we were getting to this point, it is not accounted for.